### PR TITLE
Infinite scroll

### LINKED ISSUE #88: Integrate react-infinite scroll and modal spinners.
**Is your feature request related to a problem? Please describe.**
Need optimize the way allPokemon data/cards are loaded on home page. Also, when fetching a lot of pokemons, modal tabs take time to have data fetched and displayed, so more error handling and spinner is needed.

**Describe the solution you'd like**
Use react-infinite-scroll component and spinners handling a lot of data.

### PR DESCRIPTION
**Pull request by connielion**
## Changes
1. Define `getMorePokemons` function for fetching when user scrolls. Pass it down from PokeApp to PokeList and assign it to `next` prop in `<InfiniteScroll>`.
2. Invoke `getMorePokemons` in PokeApp useEffect for initial data.
3. Added checks in PokeList render method for `allPokemons` and `filteredPokemons` lengths.

## Purpose
Add infinite scroll for fetching data.

## Approach
Use react-infinite-scroll-component.
Define new function to fetch when user scrolls.

Closes #88

### CODE REVIEW

File: src/components/PokeApp.js [hunk 1]
```diff
@@ -15,11 +15,11 @@ function PokeApp() {
   // fetchedData is a single Pokemon object the user searched for
   const [fetchedData, setFetchedData] = useState([]);
   const [allPokemons, setAllPokemon] = useState([]);
+  const [queryParams, setQueryParams]=useState({offset: 0, limit: 12});
```
Comment (Gjhnsn) [inline src/components/PokeApp.js:18]: is this fetching 12 at a time?

File: src/components/PokeApp.js [hunk 2]
```diff
@@ -40,31 +40,32 @@ function PokeApp() {
     }
   };
 
-  const getAllPokemon = async (pokeUrls) => {
-    try {
-      const pokeList = await fetchAllPokemons();
+  const getMorePokemons = async()=>{
```
Comment (dannysanchez559) [inline src/components/PokeApp.js:43]: Liking the naming updates

File: src/components/PokeApp.js [hunk 3]
```diff
@@ -40,31 +40,32 @@ function PokeApp() {
...
-        return fetchPokemon(name);
+        const pokemonData = await fetchPokemon(name);
+        return pokemonData;
       });
-
-      Promise.all(pokemonObjs).then((data) => {
-        setAllPokemon(data);
-        setFetchedData(data);
+      Promise.all(pokemonPromises).then((data) => {
+        // make copy previous state, add next 20 pokemon objects, then set state with more data
```
Comment (dannysanchez559) [inline src/components/PokeApp.js:53]: +1 on the comment

**Review by dannysanchez559 — APPROVED** [review]
Nicely done, I like how you kept up with consistent naming scheme

**Review by Gjhnsn — APPROVED** [review]
Implementing infinite scroll seems super interesting, excited to see it in action!

File: src/components/PokeApp.js (hunk 1 above)
Comment (connielion) [reply src/components/PokeApp.js:18]: Yes

**Review by BlackCubes — APPROVED** [review]
Looks really good! Hopefully you solved the issue from earlier

**Pull request merged**